backend/src/core/user/application/user-service.ts

```diff
@@ -47,7 +47,11 @@ export class UserService {
     return this.userRepository.getById(id);
   }
 
-  createUser(userRequestDto: UserSignUpRequestDto): Promise<UserSignUpResponseDto> {
+  getUserByQuery(query: Prisma.UserWhereInput): Promise<User | null> {
```

Comment:
It's not only for username and email. it's intended for all possible cases of querying (including by properties on nested documents, like UserProfile, and compound conditions like OR) so as not to spawn new methods on every demand. i'm not sure this typing would be easy to code without using Prisma pre-generated typings, and Prisma.User type was all over the app long before, so it would be needed to replace it too

Reply:
Your case covers only username and email, if someone needs to find a user by other parameters they should create a new method in UserReposytoryAdapted. The problem is that your code knows about specific ORM methods or query format if one day a customer tells you to change ORM from Prisma to Type ORM for example, it will be a huge problem to refactor

About Prisma pre-generated types better to replace them to our DTO in secondary adapters if it is possible